flink-batch-connectors/flink-avro/src/main/java/org/apache/flink/api/java/io/AvroOutputFormat.java

```diff
@@ -36,6 +38,8 @@
 	private final Class<E> avroValueType;
 
 	private transient Schema userDefinedSchema = null;
+
+	private transient CodecFactory codecFactory = null;
```

Comment:
Flink uses Java Serialization to ship code from the client to the cluster. `CodecFactory` does not implement `Serializable`. Therefore, the `codecFactory` field must be declared as `transient`. However, the `writeObject` and `readObject` methods must be adapted to manually serialize / deserialize the information. Otherwise, the `CodecFactory` is lost when the object is deserialized at the client.